user-service/model/user-orm.js

```diff
@@ -1,69 +1,37 @@
 import bcrypt from 'bcrypt';
 
-import { createUser, updateAccountByUsername, deleteAccountByUsername, getUserByUsername, getUserByEmail } from './repository.js';
+import { getUserByUsername, getUserByEmail, createUser, updateUser, deleteUser } from './user-repo.js';
 
 const saltRounds = 10;
 
-//need to separate orm functions from repository to decouple business logic from persistence
-export async function ormCreateUser(username, email, password) {
-    try {
-        bcrypt.hash(password, saltRounds, async function(err, hashedPassword) {
-            const newUser = await createUser({username, email: email.toLowerCase(), password: hashedPassword});
-            newUser.save();
-        });
-        return true;
-    } catch (err) {
-        console.log('ERROR: Could not create new user');
-        return { err };
-    }
-}
-
-// Returns the authenticated user, or null if not authenticated
-export async function authenticateUser(username, password) {
-    // Check if user exists
-    let user = await getUserByUsername(username);
-    user = user ? user : await getUserByEmail(username); // Allow signin using email too
-
-    if (!user) {
-        return null;
-    }
-
-    // Check if password is correct
-    const isMatch = await bcrypt.compare(password, user.password);
-    if (!isMatch) {
-        return null;
-    }
-    return user.toObject();
-}
-
-// Checks if username exists in database
-export async function isExistingUser(username) {
-    const user = await getUserByUsername(username);
-    return user ? true : false;
-}
+export const checkExistence = async (username, email) => {
+  if (await getUserByUsername(username) || await getUserByEmail(email.toLowerCase())) {
+    return true;
+  }
+  return false;
+};

```

Comment:
How about `return await getUserByUsername(username) || await getUserByEmail(email.toLowerCase());`

Reply:
Returning `true` or `false` here to somewhat response to the question of "Does this user exist?"

Reply:
Sure, I think the code is easier to understand and more maintainable this way